src/main/java/bio/terra/cli/command/app/passthrough/Git.java

```diff
@@ -46,15 +54,33 @@ protected void executeImpl() {
     }
     if (cloneAll) {
       validateCloneCommand();
-      Context.requireWorkspace().getResources().stream()
-          .filter(resource -> Resource.Type.GIT_REPO == resource.getResourceType())
-          .forEach(resource -> cloneGitRepoResource(resource));
+      var resources = Context.requireWorkspace().getResources();
+      getGitRepoResourceToClone(resources);
+      resources.stream()
+          .filter(resource -> Resource.Type.DATA_COLLECTION == resource.getResourceType())
+          .forEach(
+              resource -> {
+                Workspace dataCollectionWorkspace =
+                    attemptToGetDataCollectionWorkspace((DataCollection) resource);
+                if (dataCollectionWorkspace != null) {
+                  getGitRepoResourceToClone(dataCollectionWorkspace.getResources());
+                }
```

Comment:
nit: to reduce duplication, you could move this logic inside `attemptToGetDataCollectionWorkspace` and call it something like `attemptToCloneDataCollectionWorkspace` instead. that saves repeating this on lines 76-80 below

Reply:
How about

```
validateCloneCommand();

List<String> gitReposToClone;
if --all flag
  gitReposToClone = getGitReposInWorkspace()  // getGitReposInWorkspace would get DC repos
else
  gitReposToClone = get from flag

clone(gitReposToClone)
```

Reply:
we can't validate clone outside those if because it maybe another type of git command. 

combined both of your suggestions.